### ISSUE TITLE
Support Google Provider 4.0

### ISSUE THREAD

**Issue by jackwhelpton** [terraform-google-modules/terraform-google-vm#214]
### TL;DR

Module needs to support the Google provider 4.0 (to prevent problems for upstream modules).

The breaking changes related to Kubernetes/Workload Identity, so this should be as simple as updating the version constraints: PR to follow.

### Terraform Configuration

```hcl
n/a
```

### Terraform Version

```sh
Terraform v1.0.11
on windows_amd64
+ provider registry.terraform.io/hashicorp/external v2.1.0
+ provider registry.terraform.io/hashicorp/google v3.90.0
+ provider registry.terraform.io/hashicorp/google-beta v3.90.0
+ provider registry.terraform.io/hashicorp/helm v2.3.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.6.1
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/netapp/netapp-gcp v21.5.0
```

**Comment by jackwhelpton** [comment]
PR is https://github.com/terraform-google-modules/terraform-google-vm/pull/215.

**Comment by bharathkkb** [comment]
fixed by #215
